js/prefs/switchencryptionoptions.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Managing Keys for identities in the preferences UI.
+ *
+ * @author     Rafael te Boekhorst <boekhorst@b1-systems.de>
+ * @copyright  2014-2015 Horde LLC
+ * @license    GPL-2 (http://www.horde.org/licenses/gpl)
+ */
+
+var switchEncryption = {
+
+    // checks wheather the smime checkbox is clicked and shows/hides
+    checkSMIMEOption: function () {
+        // get checkbox info
+        if ($('smimeselect').checked === true) {
+            $$('.prefsSmimeContainer').invoke('show');
+        } else if ($('smimeselect').checked === false) {
+            $$('.prefsSmimeContainer').invoke('hide');
+            this.showKeys();
+        }
+
+    },
+
+    // shows the keys of the identity
+    showKeys: function () {
+
+        // get the identityID
+        let identity = $("identity");
+        let identityID = Number($F(identity));
+
+        HordeCore.doAction('getIdentityPubKey', // NOTE: methods from doAction cannot get any parameters! They expect $this->vars->$name
```

Comment:
You can/should unwrap in the Ajax layer and call into a method which actually has parameters.